lib/spack/spack/compilers/__init__.py

```diff
@@ -902,10 +902,11 @@ def _default(fn_args):
                 return value, None
 
             error = f"Couldn't get version for compiler {path}".format(path)
-        except spack.util.executable.ProcessError as e:
-            error = f"Couldn't get version for compiler {path}\n" + str(e)
+        # ProcessTimeoutError is a subclass of ProcessError, needs to be caught first
         except spack.util.executable.ProcessTimeoutError as e:
             error = f"Couldn't get version for compiler {path}\n" + str(e)
+        except spack.util.executable.ProcessError as e:
+            error = f"Couldn't get version for compiler {path}\n" + str(e)
```

Comment:
(minor) I don't get this. One is a subclass of the other. The error message produced seems to be exactly the same. Why are we not deleting the subclass?